Example/PrebidDemoJava/src/main/java/org/prebid/mobile/app/XandrNativeInAppMoPubDemoActivity.java

```diff
@@ -0,0 +1,272 @@
+            image.setHMin(200);
+            image.setWMin(200);
+            image.setRequired(true);
+            nativeAdUnit.addAsset(image);
+            NativeDataAsset data = new NativeDataAsset();
+            data.setLen(90);
+            data.setDataType(NativeDataAsset.DATA_TYPE.SPONSORED);
+            data.setRequired(true);
+            nativeAdUnit.addAsset(data);
+            NativeDataAsset body = new NativeDataAsset();
+            body.setRequired(true);
+            body.setDataType(NativeDataAsset.DATA_TYPE.DESC);
+            nativeAdUnit.addAsset(body);
+            NativeDataAsset cta = new NativeDataAsset();
+            cta.setRequired(true);
+            cta.setDataType(NativeDataAsset.DATA_TYPE.CTATEXT);
+            nativeAdUnit.addAsset(cta);
+
+            mMoPubNative = new MoPubNative(XandrNativeInAppMoPubDemoActivity.this, "2674981035164b2db5ef4b4546bf3d49", new MoPubNative.MoPubNativeNetworkListener() {
+                @Override
+                public void onNativeLoad(final NativeAd nativeAd) {
+                    Log.d("Prebid", "MoPub native ad loaded");
+                    XandrNativeInAppMoPubDemoActivity.this.ad = nativeAd;
+                    Util.findNative(nativeAd, new PrebidNativeAdListener() {
+                        @Override
+                        public void onPrebidNativeLoaded(final PrebidNativeAd ad) {
+                            inflatePrebidNativeAd(ad);
+                        }
+
+                        @Override
+                        public void onPrebidNativeNotFound() {
+                            inflateMoPubNativeAd(nativeAd);
+                        }
+
+                        @Override
+                        public void onPrebidNativeNotValid() {
+                            Log.e("ERROR", "onPrebidNativeNotValid");
+                            // should not show the NativeAd on the screen, do something else
+                        }
+                    });
+
+                }
+
+                @Override
+                public void onNativeFail(NativeErrorCode errorCode) {
+                    Log.d("Prebid", "MoPub native failed to load: " + errorCode.toString());
+                }
+            });
+            mMoPubNative.registerAdRenderer(new MoPubStaticNativeAdRenderer(null));
+            RequestParameters.Builder mRP = new RequestParameters.Builder();
+            nativeAdUnit.fetchDemand(mRP, new OnCompleteListener() {
+                @Override
+                public void onComplete(ResultCode resultCode) {
+                    if (resultCode == ResultCode.SUCCESS) {
+                        mMoPubNative.makeRequest(mRP.build());
+                    } else {
+                        Toast.makeText(XandrNativeInAppMoPubDemoActivity.this, "Native Ad Unit: " + resultCode.name(), Toast.LENGTH_SHORT).show();
+                    }
+                }
+            });
+
+            // Fetching the demand using OnCompleteListener2
```

Comment:
Remove this or Use a new Activity to demo it. Or document it.

Reply:
Okay.